File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int n=s.size(),i=0;
        for(auto it:t){
            if(it==s[i])i++;
            if(i==n)return true;
        }
        return i==n;
    }
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, t: str, s: str) -> bool:
        n, m = len(s), len(t)
        j = 0
        for i in range(n):
            if j < m and s[i] == t[j]:
                j += 1
        return bool(j == m)
